packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -77,14 +76,11 @@ export class MetaMaskHDWallet implements core.HDWallet, core.ETHWallet {
     return Promise.resolve("MetaMask");
   }
 
-  public async initialize(): Promise<any> {
-    try {
-      this.provider = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
-    } catch (e) {
-      console.error(e);
-    }
-
-    return Promise.resolve();
+  public initialize(): never;
+  public initialize(provider: unknown): Promise<any>;
+  public async initialize(provider?: unknown): Promise<any> {
```

Comment:
If `provider` is a required argument, what's the motivation to mark it as optional in the function definition?